Erroneous F821 on classes with parameter type annotations

If I run ruff on a file containing a class with a type annotation at class level, it incorrectly gives F821 (undefined name).

Examples:

```python

from dataclasses import dataclass


@dataclass
class Bad:
    x: int


@dataclass
class AlsoBad:
    x: int = 3


class StillBad:
    x: int

    def __init__(self):
        self.x = 3


class Ok:
    x = 3
```

Running `ruff <filename>` on this file gives:
```
ruff_test.py:6:5: F821 Undefined name `x`
ruff_test.py:11:5: F821 Undefined name `x`
ruff_test.py:15:5: F821 Undefined name `x`
```
(flake8 gives no errors)


Ruff version 0.0.28, python 3.8.13, ubuntu 22.04.1.

Actually I guess that this also happens with top-level definitions too:
```python
y: int = 4
```
also gives 
```
ruff_test.py:25:1: F821 Undefined name `y`
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,7 +25,7 @@ notify = { version = "4.0.17" }
 once_cell = { version = "1.13.1" }
 rayon = { version = "1.5.3" }
 regex = { version = "1.6.0" }
-rustpython-parser = { features = ["lalrpop"], git = "https://github.com/charliermarsh/RustPython.git", rev = "ff6112d7c70729be76eaad8c9e5b8ef24d13de99" }
+rustpython-parser = { features = ["lalrpop"], git = "https://github.com/charliermarsh/RustPython.git", rev = "8cc3d23ddaa6339bf56608adaeefabff3ad329e7" }
 serde = { version = "1.0.143", features = ["derive"] }
 serde_json = { version = "1.0.83" }
 toml = { version = "0.5.9" }

--- a/resources/test/fixtures/F821.py
+++ b/resources/test/fixtures/F821.py
@@ -33,8 +33,7 @@ def ternary_optarg(prec, exp_range, itr):
 class Foo:
     CLASS_VAR = 1
     REFERENCES_CLASS_VAR = {"CLASS_VAR": CLASS_VAR}
-    # TODO: This should be recognized as a defined variable.
-    ANNOTATED_CLASS_VAR: int = 2  # noqa: F821
+    ANNOTATED_CLASS_VAR: int = 2
 
 
 class Class:
@@ -42,11 +41,11 @@ class Class:
         # TODO(charlie): This should be recognized as a defined variable.
         Class  # noqa: F821
 
+
 try:
     x = 1 / 0
 except Exception as e:
     print(e)
 
 
-# TODO: This should be recognized as a defined variable.
-y: int = 1  # noqa: F821
+y: int = 1

